server/src/main/java/org/elasticsearch/action/ActionModule.java

```diff
@@ -545,37 +546,68 @@ public ActionModule(
                 new RestHeaderDefinition(Task.X_ELASTIC_PRODUCT_ORIGIN_HTTP_HEADER, false)
             )
         ).collect(Collectors.toSet());
-        UnaryOperator<RestHandler> restInterceptor = null;
+        UnaryOperator<RestHandler> restInterceptor = getRestServerComponent(
+            "REST interceptor",
+            actionPlugins,
+            restPlugin -> restPlugin.getRestHandlerInterceptor(threadPool.getThreadContext())
+        );
+        mappingRequestValidators = new RequestValidators<>(
+            actionPlugins.stream().flatMap(p -> p.mappingRequestValidators().stream()).toList()
+        );
+        indicesAliasesRequestRequestValidators = new RequestValidators<>(
+            actionPlugins.stream().flatMap(p -> p.indicesAliasesRequestValidators().stream()).toList()
+        );
+        headersToCopy = headers;
+
+        var customController = getRestServerComponent(
```

Comment:
the plan is to support multiple API Versions on the same cluster, right?
isn't the limit of one plugin implementing RestServerActionPlugin prevent us to do this?
When reading the doc I was imagining we would like to provide multiple RestControllers - one per one Version supported.
how will customController support multiple API versions?

Reply:
Right now we're only supporting 1 API version, so that's a future problem.
However, I'd expect it to be a single rest controller in that case with some additional logic for dispatch - similar to how the existing controller handles V7 endpoints.